### PR TITLE
feat: 🚀  TemplateAPI 和 GeneratorAPI 整合，抽象为一个父类，并新增 Template 生成逻辑

### LINKED ISSUE #149: TemplateAPI 内容整理，部分函数冗余或不符合语义需要修正

### PR DESCRIPTION
**Pull request by yhstsy**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/xun082/create-neat/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[√ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

- Closes #149 
- Related to #158 

## What is the new behavior?

TemplateAPI 和 GeneratorAPI 整合，抽象为一个父类，并新增 Template 生成逻辑

## Does this PR introduce a breaking change?

```
[ ] Yes
[√ ] No
```

### CODE REVIEW

File: packages/core/src/models/Generator.ts [hunk 1]
```diff
@@ -197,6 +198,58 @@ class Generator {
     }
   }
 
+  // 单独处理一个框架相关文件
```
Comment (uaenaTzx) [inline packages/core/src/models/Generator.ts:201]: 这一块不要写死吧，而且和 pluginGenerate 有很大程度的耦合，看看能不能解藕操作一下？

File: packages/core/src/models/BaseAPI.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      delete this.packageData[optionsType][name];
+    }
+  }
+
+  /**
+   * 扩展项目的 package.json
+   * @param fields 合并内容
+   * @param {object} [options] 操作选项
+   */
+  extendPackage(fields: object, options: object = {}) {
```
Comment (xun082) [inline packages/core/src/models/BaseAPI.ts:54]: ts类型尽量别写Object

File: packages/core/src/models/BaseAPI.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      if (!extendOptions.merge || !(key in pkg)) {
+        pkg[key] = value;
+      } else if (Array.isArray(value) && Array.isArray(existing)) {
+        // 如果是数组则合并 且去重
+        pkg[key] = existing.concat(value.filter((v) => existing.indexOf(v) < 0));
+      } else if (typeof value === "object" && typeof existing === "object") {
+        // 如果是对象则合并
+        pkg[key] = { ...existing, ...value };
+      } else {
+        pkg[key] = value;
```
Comment (xun082) [inline packages/core/src/models/BaseAPI.ts:85]: 对象判断严谨一点，万一是个null呢，用typeof不太严谨

File: packages/core/src/models/BaseAPI.ts (hunk 2 above)
Comment (yhstsy) [reply packages/core/src/models/BaseAPI.ts:54]: 收到✔

File: packages/core/src/models/BaseAPI.ts (hunk 3 above)
Comment (yhstsy) [reply packages/core/src/models/BaseAPI.ts:85]: 收到✔

**Review by uaenaTzx — APPROVED** [review]

**Pull request merged**